sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShuffleBufferCatalog.scala

```diff
@@ -16,52 +16,77 @@
 
 package com.nvidia.spark.rapids
 
-import java.io.File
 import java.util.concurrent.ConcurrentHashMap
 import java.util.concurrent.atomic.AtomicInteger
 import java.util.function.{Consumer, IntUnaryOperator}
 
 import scala.collection.mutable.ArrayBuffer
 
-import ai.rapids.cudf.{ContiguousTable, Cuda, DeviceMemoryBuffer}
+import ai.rapids.cudf.{ContiguousTable, Cuda, DeviceMemoryBuffer, Table}
 import com.nvidia.spark.rapids.Arm.withResource
 import com.nvidia.spark.rapids.format.TableMeta
+import com.nvidia.spark.rapids.spill.{SpillableDeviceBufferHandle, SpillableHandle}
 
-import org.apache.spark.SparkEnv
+import org.apache.spark.{SparkEnv, TaskContext}
 import org.apache.spark.internal.Logging
-import org.apache.spark.sql.rapids.RapidsDiskBlockManager
 import org.apache.spark.sql.rapids.execution.TrampolineUtil
+import org.apache.spark.sql.types.DataType
+import org.apache.spark.sql.vectorized.ColumnarBatch
 import org.apache.spark.storage.ShuffleBlockId
 
 /** Identifier for a shuffle buffer that holds the data for a table */
 case class ShuffleBufferId(
     blockId: ShuffleBlockId,
-    override val tableId: Int) extends RapidsBufferId {
+    tableId: Int) {
   val shuffleId: Int = blockId.shuffleId
   val mapId: Long = blockId.mapId
-
-  override val canShareDiskPaths: Boolean = true
-
-  override def getDiskPath(diskBlockManager: RapidsDiskBlockManager): File = {
-    diskBlockManager.getFile(blockId)
-  }
 }
 
 /** Catalog for lookup of shuffle buffers by block ID */
-class ShuffleBufferCatalog(
-    catalog: RapidsBufferCatalog,
-    diskBlockManager: RapidsDiskBlockManager) extends Logging {
+class ShuffleBufferCatalog extends Logging {
+  /**
+   * Information stored for each active shuffle.
+   * NOTE: ArrayBuffer in blockMap must be explicitly locked when using it!
+   *
+   * @param blockMap mapping of block ID to array of buffers for the block
+   */
+  private case class ShuffleInfo(
```

Comment:
nit: there is a single value in this class. Would it be better to just alias ShuffleInfo to The ConcurrentHashMap...